### PR TITLE
#2: add query API

### PR DESCRIPTION
**Pull request by lweides**

### CODE REVIEW

File: src/main/java/column/store/api/column/IdColumn.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package column.store.api.column;
+
+public final class IdColumn extends BaseColumn {
```
Comment (Eliasrpx) [inline src/main/java/column/store/api/column/IdColumn.java:3]: I assume this is only used for explicitly identifying related column values ?

File: src/main/java/column/store/api/query/DoubleFilter.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+         */
+        public DoubleFilter isGreaterThan(double minExclusive) {
+            return new DoubleFilter(column, MatchType.GREATER_THAN, minExclusive, Double.MAX_VALUE);
+        }
+
+        /**
+         * @return a new {@link DoubleFilter}, which matches a record iff {@code minInclusive <= record[column] < maxExclusive}.
+         */
+        public DoubleFilter isBetween(double minInclusive, double maxExclusive) {
+            return new DoubleFilter(column, MatchType.BETWEEN, minInclusive, maxExclusive);
```
Comment (Eliasrpx) [inline src/main/java/column/store/api/query/DoubleFilter.java:74]: Should probably check for a valid range (i.e.: isBetween(10, 9) -> should not be possible)

File: src/main/java/column/store/api/query/LongFilter.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+         */
+        public LongFilter isGreaterThan(long minExclusive) {
+            return new LongFilter(column, MatchType.GREATER_THAN, minExclusive, Long.MAX_VALUE);
+        }
+
+        /**
+         * @return a new {@link LongFilter}, which matches a record iff {@code minInclusive <= record[column] < maxExclusive}.
+         */
+        public LongFilter isBetween(long minInclusive, long maxExclusive) {
+            return new LongFilter(column, MatchType.BETWEEN, minInclusive, maxExclusive);
```
Comment (Eliasrpx) [inline src/main/java/column/store/api/query/LongFilter.java:74]: same as [above](https://github.com/lweides/column-store/pull/7/commits/8c818c963a050a79f8694af1405615616b61087e#r1363366952)

File: src/main/java/column/store/api/query/Query.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    Iterable<Column> columns();
+
+    /**
+     * @return the {@link QueryType} of the {@link Query}.
+     */
+    QueryType type();
+
+    enum QueryType {
+        ALL_OFF,
```
Comment (Eliasrpx) [inline src/main/java/column/store/api/query/Query.java:36]: typo: ALL_OFF -> you mean ALL_OF ?

File: src/main/java/column/store/api/column/IdColumn.java (hunk 1 above)
Comment (lweides) [reply src/main/java/column/store/api/column/IdColumn.java:3]: A column for ids, i.e. user-ids, etc. Values which are inherently random, and therefore compress bad.

File: src/main/java/column/store/api/query/DoubleFilter.java (hunk 2 above)
Comment (lweides) [reply src/main/java/column/store/api/query/DoubleFilter.java:74]: Good point, will add 👍

File: src/main/java/column/store/api/query/LongFilter.java (hunk 3 above)
Comment (lweides) [reply src/main/java/column/store/api/query/LongFilter.java:74]: Yep

File: src/main/java/column/store/api/query/Query.java (hunk 4 above)
Comment (lweides) [reply src/main/java/column/store/api/query/Query.java:36]: Yep

**Review by Eliasrpx — APPROVED** [review]

**Pull request merged**